pkg/util/resource.go

```diff
@@ -68,3 +68,22 @@ func IsResourceDiff(old, new corev1.ResourceList, resourceName corev1.ResourceNa
 
 	return newQuant >= oldQuant*(1+diffThreshold) || newQuant <= oldQuant*(1-diffThreshold)
 }
+
+// IsequalTopo returns whether the new topology has difference with the old one or not
+func IsequalTopo(oldtopo map[string]string, newtopo map[string]string) bool {
+	keySlice := make([]string, 0)
```

Comment:
This check for equality may not be sufficient. For example, what is recorded in the annotation is a JSON object, but the serialization of the JSON object does not guarantee order (at this time, the JSON object data itself does not change.)